nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/PutCassandraRecord.java

```diff
@@ -129,8 +210,39 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         final String cassandraTable = context.getProperty(TABLE).evaluateAttributeExpressions(inputFlowFile).getValue();
         final RecordReaderFactory recordParserFactory = context.getProperty(RECORD_READER_FACTORY).asControllerService(RecordReaderFactory.class);
         final int batchSize = context.getProperty(BATCH_SIZE).evaluateAttributeExpressions().asInteger();
-        final String batchStatementType = context.getProperty(BATCH_STATEMENT_TYPE).getValue();
         final String serialConsistencyLevel = context.getProperty(CONSISTENCY_LEVEL).getValue();
+        final String updateKeys = context.getProperty(UPDATE_KEYS).evaluateAttributeExpressions(inputFlowFile).getValue();
```

Comment:
if the `Update Keys` property is not set, then `updateKeys` is `null`, not an empty string as is set in the unit tests. This causes a NullPointerException where the update keys are split from the property value. A null check should be added, either to set the field to the empty string or to throw a different exception than NPE. This can/should also be done in `customValidate`, which illustrates the need for custom validation code (albeit duplicated at times).

Reply:
I've added a unit test for this, and added a customValidate to check it.